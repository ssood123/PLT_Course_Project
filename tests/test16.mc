float i;

function void main()
{
  int i;
  i = 7;
  print(i);
}
