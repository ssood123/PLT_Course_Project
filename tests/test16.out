; ModuleID = 'integraph'
source_filename = "integraph"

@i = global double 0.000000e+00
@fmt = private unnamed_addr constant [4 x i8] c"%d\0A\00", align 1
@fmt.1 = private unnamed_addr constant [4 x i8] c"%s\0A\00", align 1
@fmt.2 = private unnamed_addr constant [4 x i8] c"%g\0A\00", align 1

declare i32 @printf(i8*, ...)

declare i32 @printbig(i32)

define void @main() {
entry:
  %i = alloca i32, align 4
  store i32 7, i32* %i, align 4
  %i1 = load i32, i32* %i, align 4
  %printf = call i32 (i8*, ...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @fmt, i32 0, i32 0), i32 %i1)
  ret void
}
